src/tactic/linear_combination.lean

```diff
@@ -266,6 +266,19 @@ This tactic only should be used when the target's type is an equality whose
 meta def set_goal_to_hleft_sub_tleft (hsum_on_left : expr) : tactic unit :=
 do to_expr ``(eq_zero_of_sub_eq_zero %%hsum_on_left) >>= apply, skip
 
+/--
+If an exponent `n` is provided, changes the goal from `t = 0` to `t^n = 0`.
+* Input:
+  * `exponent : option ℕ`, the power to raise the goal by. If `none`, this tactic is a no-op.
+
+* Output: N/A
+-/
+meta def raise_goal_to_power (exponent : option ℕ) : tactic unit :=
+match exponent with
+| some n := refine ``(@pow_eq_zero _ _ _ _ %%`(n) _)
+| none := skip
+end
```

Comment:
What do you think of
```suggestion
meta def raise_goal_to_power (exponent : ℕ) : tactic unit :=
match exponent with
| 1 := skip
| n := refine ``(@pow_eq_zero _ _ _ _ %%`(n) _)
end
```
This has the behavior that `linarith with exponent 1` works even without `no_zero_divisors R`, and you don't need a special out-of-band value for "don't exponentiate".

Reply:
Good call!